### DISCUSSION TITLE
libphp.so Error When Trying To Start Apache

### DISCUSSION THREAD

**Discussion by brentgb** [Homebrew/discussions#1015] [Getting started] [answered]
I installed Apache and PHP successfully via Homebrew on my MacBook Air M1/Apple Silicon machine several weeks ago.  Apache would start and PHP would load and I could view everything at http://localhost:8080/ in my browser as expected.

A few days ago, I ran brew update and brew upgrade and updates were installed.  Today, when I tried to start Apache, I get this error:
```
httpd: Syntax error on line 534 of /opt/homebrew/etc/httpd/httpd.conf: Cannot load /opt/homebrew/opt/php/lib/httpd/modules/libphp.so into server: dlopen(/opt/homebrew/opt/php/lib/httpd/modules/libphp.so, 10): Library not loaded: /opt/homebrew/opt/icu4c/lib/libicuio.67.dylib\n  Referenced from: /opt/homebrew/opt/php/lib/httpd/modules/libphp.so\n  Reason: image not found
```

I was curious if anyone else has had this problem and, if so, if they knew how to fix it.

I've tried changing the path in httpd.conf to what appear to be new (for instance, this was apparently created a few days ago after upgrading: /opt/homebrew/Cellar/php/8.0.2/lib/httpd/modules/libphp.so) locations to the libphp.so file, but I get a similar error.

Thanks for any help.

**Comment by SMillerDev** [comment, marked as answer]
Try `brew reinstall icu4c`

**Reply by carlocab** [reply]
Doing `brew update && brew upgrade` before that is probably a good idea.

**Reply by brentgb** [reply]
Thank you!  I did what you both suggested - I got an error about needing to install Apple Command Line Tools, which I swear I installed when I first set up Homebrew and installed PHP, Apache, etc.  However, I installed them, then ran brew update, brew upgrade, and brew reinstall icu4c, started Apache and it worked.  Thank you again for your help!

**Reply by carlocab** [reply]
Did you run an OS update recently?

Even minor version updates get rid of CLT installs now. It's inconvenient, but right now you'll need to remember to do `xcode-select --install` each time you update your OS.

**Reply by brentgb** [reply]
Yes, I did - 11.2.2.  I also installed Xcode 12.  Thanks for the heads-up!
